demo-shell/src/app/components/app-layout/app-layout.component.ts

```diff
@@ -48,6 +50,25 @@ export class AppLayoutComponent {
         { href: '/about', icon: 'info_outline', title: 'APP_LAYOUT.ABOUT' }
     ];
 
-    constructor() {
+    expandedSidenav: Boolean = false;
+
+
+    ngOnInit() {
+        const expand = this.config.get<boolean>('sideNav.expandedSidenav');
+        const preserveState = this.config.get('sideNav.preserveState');
+
+        if (preserveState && expand)  {
+            this.expandedSidenav = (this.userpreference.get('expandedSidenav', String(expand)) === 'true');
```

Comment:
I'm confused about this String expression here.
according to how the "expand" is retrieved I presume, the expand is already a boolean (this.config.get&lt;boolean&gt;). Why don't we just compare it to a boolean value, why is the casting needed?
expand === true

Reply:
The get method from user preference service require a string param, and also returns a string, that's why I did the casting.